components/Buttons/tests/unit/MDCButtonTests.m

```diff
@@ -590,6 +590,77 @@ - (void)testTitleFontForStateWithLegacyDynamicTypeRestoresUserFonts {
                 self.button.titleLabel.font);
 }
 
+- (void)testTitleFontForStateDisabledAfterSettingFontsPreventsFontChange {
+  // Given
+  UIFont *normalFont = [UIFont systemFontOfSize:10];
+  UIFont *selectedFont = [UIFont systemFontOfSize:40];
+  UIFont *directlyAssignedFont = [UIFont systemFontOfSize:25];
+  [self.button setTitleFont:normalFont forState:UIControlStateNormal];
+  [self.button setTitleFont:selectedFont forState:UIControlStateSelected];
+
+  // When
+  self.button.enableTitleFontForState = NO;
+  self.button.titleLabel.font = directlyAssignedFont;
+  self.button.selected = YES;
+
+  // Then
+  XCTAssertEqualObjects(self.button.titleLabel.font, directlyAssignedFont);
+}
+
+- (void)testTitleFontForStateDisabledBeforeSettingFontsPreventsFontChange {
+  // Given
+  UIFont *normalFont = [UIFont systemFontOfSize:10];
+  UIFont *selectedFont = [UIFont systemFontOfSize:40];
+  UIFont *directlyAssignedFont = [UIFont systemFontOfSize:25];
+  self.button.enableTitleFontForState = NO;
+  self.button.titleLabel.font = directlyAssignedFont;
+
+  // When
+  [self.button setTitleFont:normalFont forState:UIControlStateNormal];
+  [self.button setTitleFont:selectedFont forState:UIControlStateSelected];
+  self.button.selected = YES;
+
+  // Then
+  XCTAssertEqualObjects(self.button.titleLabel.font, directlyAssignedFont);
+}
+
+- (void)testTitleFontForStateReenabledDoesNotImmediatelyUpdateFont {
```

Comment:
can you elaborate why its important that we do not update the font immediately once we set this flag?

Reply:
It's not important, but it seems unexpected. If you want, I can change the behavior.

Reply:
Adding a test for behavior makes me think it is something we want to preserve. I would change the behavior.

Reply:
Ok, thanks. I'll remove the test. I wouldn't expect the font to change immediately.